### PR TITLE
Repository: add compact_segments progress

### LINKED ISSUE #1115: no progress bar at the segment cleanup step for delete/prune
when running a delete on a no-append-only repository,
after the decrement references step, there is a cleanup step, which does not show a progress bar

so on a large repo one is left wondering

### PR DESCRIPTION
**Pull request by enkore**
Simple, segment based, so depending on max_segment_size more or less granular.

Fixes #1115

### CODE REVIEW

**Comment by enkore** [comment]
merge?

File: src/borg/testsuite/archiver.py [hunk 1]
```diff
@@ -228,6 +228,8 @@ def tearDown(self):
         os.chdir(self._old_wd)
         # note: ignore_errors=True as workaround for issue #862
         shutil.rmtree(self.tmpdir, ignore_errors=True)
+        # destroy logging configuration
+        logging.Logger.manager.loggerDict.clear()
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: how is that related?

Comment (enkore) [inline src/borg/testsuite/archiver.py]: state leakage in the test i split in two. --progress creates some logging configuration that is carried over between invocations (self.cmd) and tests themselves in the `logging` module.
